wallet/chainntfns.go

```diff
@@ -888,18 +895,26 @@ func (w *Wallet) handleWinningTickets(blockHash *chainhash.Hash,
 	// be fixed somehow, but this should be stable for networks that
 	// are voting at normal block speeds.
 
-	// Only consider tickets owned by this wallet.
-	ticketHashes := w.StakeMgr.OwnTickets(winningTicketHashes)
-	if len(ticketHashes) == 0 {
-		return nil
-	}
-
-	votes := make([]*wire.MsgTx, len(ticketHashes))
+	var ticketHashes []*chainhash.Hash
+	var votes []*wire.MsgTx
 	voteBits := w.VoteBits()
 	err = walletdb.View(w.db, func(dbtx walletdb.ReadTx) error {
+		txmgrNs := dbtx.ReadBucket(wtxmgrNamespaceKey)
+
+		// Only consider tickets owned by this wallet.
+		ticketHashes = selectOwnedTickets(w, dbtx, winningTicketHashes)
+		if len(ticketHashes) == 0 {
+			return nil
+		}
+
+		votes = make([]*wire.MsgTx, len(ticketHashes))
+
 		addrmgrNs := dbtx.ReadBucket(waddrmgrNamespaceKey)
 		for i, ticketHash := range ticketHashes {
-			ticketPurchase, err := w.StakeMgr.TicketPurchase(dbtx, ticketHash)
+			ticketPurchase, err := w.TxStore.Tx(txmgrNs, ticketHash)
+			if err != nil || ticketPurchase == nil {
```

Comment:
It's doing this for backwards compat with stakepools that may rely on stakemgr working as before.  Tickets could be manually added to the wallet and would be used for voting, but those ticket only were saved in stakemgr and would not be found by the Tx call.  Tx returns nil if the transaction isn't found, so in this case it would have to be looked up by stakemgr.